src/main/java/org/broadinstitute/hellbender/metrics/PerUnitInsertSizeMetricsCollector.java

```diff
@@ -0,0 +1,226 @@
+package org.broadinstitute.hellbender.metrics;
+
+import htsjdk.samtools.SamPairUtil;
+import htsjdk.samtools.util.Histogram;
+import htsjdk.samtools.metrics.MetricsFile;
+
+import java.io.Serializable;
+import java.util.*;
+
+/**
+ * A Collector for individual InsertSizeMetrics for a given SAMPLE or SAMPLE/LIBRARY or
+ * SAMPLE/LIBRARY/READ_GROUP (depending on aggregation levels)
+ */
+public final class PerUnitInsertSizeMetricsCollector
+        implements PerUnitMetricCollector<InsertSizeMetrics, Integer, InsertSizeMetricsCollectorArgs>,
+                    Serializable
+{
+    private static final long serialVersionUID = 1L;
+
+    // kryo can't serialize an EnumMap so use a LinkedHashMap (also so we can maintain order
+    // the insertion order of entries since the combining code below counts on it
+    final Map<SamPairUtil.PairOrientation, Histogram<Integer>> Histograms = new LinkedHashMap<>();
+
+    final String sample;
+    final String library;
+    final String readGroup;
+
+    // When generating the Histogram, discard any data categories (out of FR, TANDEM, RF) that have fewer than this
+    // percentage of overall reads. (Range: 0 to 1)
+    private final double minimumPct;
+
+    // Generate mean, sd and plots by trimming the data down to MEDIAN + DEVIATIONS*MEDIAN_ABSOLUTE_DEVIATION.
+    // This is done because insert size data typically includes enough anomolous values from chimeras and other
+    // artifacts to make the mean and sd grossly misleading regarding the real distribution.
+    private final double deviations;
+
+    //Explicitly sets the Histogram width, overriding automatic truncation of Histogram tail.
+    //Also, when calculating mean and stdev, only bins <= HISTOGRAM_WIDTH will be included.
+    private Integer HistogramWidth;
```

Comment:
This value comes from an optional command line argument that may be specified by the user. Added a comment.